cartoframes/viz/legend.py

```diff
@@ -42,35 +42,52 @@ def _init_legend(self, data):
                 self._title = data.get('title', '')
                 self._description = data.get('description', '')
                 self._footer = data.get('footer', '')
-
-                if self._type or self._prop:
-                    if not isinstance(self._type, dict) and self._type not in constants.LEGEND_TYPES:
-                        raise ValueError(
-                            'Legend type "{0}" is not valid. Valid legend types are: {1}'.format(
-                                self._type,
-                                ', '.join(constants.LEGEND_TYPES)
-                            ))
-                    if self._prop not in constants.LEGEND_PROPERTIES:
-                        raise ValueError(
-                            'Legend property "{0}" is not valid. Valid legend property are: {1}'.format(
-                                self._prop,
-                                ', '.join(constants.LEGEND_PROPERTIES)
-                            ))
-
             else:
-                raise ValueError('Wrong legend input')
+                raise ValueError('Wrong legend input.')
 
-    def get_info(self, geom_type):
-        if (self._type and self._prop) or self._title or self._description or self._footer:
+    def get_info(self, geom_type=None):
+        if self._type or self._title or self._description or self._footer:
             _type = self._type
+            _prop = self._prop
+
             if isinstance(_type, dict) and geom_type in _type:
                 _type = _type.get(geom_type)
+            self._check_type(_type)
+
+            if _type and not _prop:
+                _prop = self._infer_prop(_type)
+
+            self._check_prop(_prop)
+
+            print(_prop)
+
             return {
                 'type': _type,
-                'prop': self._prop,
+                'prop': _prop,
                 'title': self._title,
                 'description': self._description,
                 'footer': self._footer
             }
         else:
             return {}
+
+    def _check_type(self, _type):
+        if _type and _type not in constants.LEGEND_TYPES:
+            raise ValueError(
+                'Legend type is not valid. Valid legend types are: {}.'.format(
+                    ', '.join(constants.LEGEND_TYPES)
+                ))
+
+    def _check_prop(self, _prop):
+        print(_prop)
```

Comment:
Remove `print`

Reply:
As an improvement, I'd move `_type` and `_prop` initialization to separated methods and I'd leave the responsibility of checking if they're correct to these methods:

```python
def _get_type(self, geom_type):
    _type = self._type.get(geom_type)
        if isinstance(self._type, dict) and geom_type in self._type
        else self._type
  
    self._check_type(self._type)
    return _type
```

```python
def _get_prop(self, _type):
    _prop = self._prop if self._prop else self._infer_prop(_type)
    
    self._check_prop(_prop)
    return _prop
```

And then, in `get_info`:

```python
def get_info(self, geom_type=None):
   # ...
    _type = self._get_type(geom_type)
    _prop = self._get_prop(_type)
```